app/assets/javascripts/govuk_publishing_components/analytics/explicit-cross-domain-links.js

```diff
@@ -1,89 +1,88 @@
+    } else {
+      attributeValue += '?' + param
     }
 
-    this.handleCookiesAccepted = function () {
-      // If the cookie banner was engaged and usage cookie accepted, append ?_ga=clientid if available and cookie_consent=accept
-      var element = this.element
-      var attribute = this.attribute
-      this.decorate(element, 'cookie_consent=accept', attribute)
+    element.setAttribute(attribute, attributeValue)
+  }
 
-      if (!global.ga) {
-        return
-      }
+  ExplicitCrossDomainLinks.prototype.handleEvent = function (e) {
+    //  prevent default: we want the link href and/or form action to be decorated before we navigate away
+    e.preventDefault()
+    var cookieBannerEngaged = window.GOVUK.cookie('cookies_preferences_set')
+    var cookieConsent = window.GOVUK.getConsentCookie()
+
+    if (cookieBannerEngaged !== 'true') {
+      // If not engaged, append only ?cookie_consent=not-engaged
+      this.decorate(this.$module, 'cookie_consent=not-engaged', this.attribute)
+    } else if (cookieConsent && cookieConsent.usage === true) {
+      this.handleCookiesAccepted()
+    } else {
+      this.decorate(this.$module, 'cookie_consent=reject', this.attribute)
+    }
 
-      global.ga(function () {
-        var trackers = global.ga.getAll()
+    // remove the event listener to avoid an infinite loop
+    this.$module.removeEventListener(this.eventType, this.handleEvent)
 
-        if (!trackers.length) { return }
+    // if the element is a form, submit it. If it is a link, click it
+    if (this.eventType === 'submit') {
+      this.$module.submit()
+    } else {
+      this.$module.click()
+    }
+  }
 
-        var linker = new global.gaplugins.Linker(trackers[0])
-        var attrValue = element.getAttribute(attribute)
+  ExplicitCrossDomainLinks.prototype.handleCookiesAccepted = function () {
+    // If the cookie banner was engaged and usage cookie accepted, append ?_ga=clientid if available and cookie_consent=accept
+    var element = this.$module
+    var attribute = this.attribute
+    this.decorate(element, 'cookie_consent=accept', attribute)
 
-        element.setAttribute(attribute, linker.decorate(attrValue))
-      })
+    if (!window.ga) {
+      return
     }
 
-    this.decorate = function (element, param, attribute) {
-      var attributeValue = element.getAttribute(attribute)
+    window.ga(function () {
+      var trackers = window.ga.getAll()
 
-      if (!attributeValue) { return }
+      if (!trackers.length) { return }
 
-      if (attributeValue.indexOf('?') !== -1) {
-        attributeValue += '&' + param
-      } else {
-        attributeValue += '?' + param
-      }
+      var linker = new window.gaplugins.Linker(trackers[0])
+      var attrValue = element.getAttribute(attribute)
 
-      element.setAttribute(attribute, attributeValue)
+      element.setAttribute(attribute, linker.decorate(attrValue))
+    })
+  }
+
+  ExplicitCrossDomainLinks.prototype.init = function () {
```

Comment:
Tiny nitpick - could you put the `init` function as the first function? Just because we do that everywhere else and it's the first bit of code that executes.